File: examples/ruby/exponentiation_exponent_then_mod.rb
# Ported from examples/lumen/exponentiation_exponent_then_mod.lm by scripts/port_examples.py; edit the Lumen original, not this file.
base = 7
exp = 100
mod = 1000000007
iterations = 100
puts("Fast modular exponentiation benchmark")
print("base = ")
puts(base)
print("exp  = ")
puts(exp)
print("mod  = ")
puts(mod)
print("iterations = ")
puts(iterations)
puts("")
puts("Running mod_pow...")
result = 0
i = 0
while i < iterations do
    result = mod_pow(base, exp, mod)
    i = i + 1
end
print("Result: ")
puts(result)
puts("Done!")

File: examples/c/exponentiation_exponent_then_mod.c
// Ported from examples/lumen/exponentiation_exponent_then_mod.lm by scripts/port_examples.py; edit the Lumen original, not this file.
int main(void) {
    long base = 7;
    long exp = 100;
    long mod = 1000000007;
    long iterations = 100;
    puts("Fast modular exponentiation benchmark");
    printf("base = ");
    printf("%ld\n", base);
    printf("exp  = ");
    printf("%ld\n", exp);
    printf("mod  = ");
    printf("%ld\n", mod);
    printf("iterations = ");
    printf("%ld\n", iterations);
    puts("");
    puts("Running mod_pow...");
    long result = 0;
    long i = 0;
    while (i < iterations) {
        result = mod_pow(base, exp, mod);
        i = i + 1;
    }
    printf("Result: ");
    printf("%ld\n", result);
    puts("Done!");
    return 0;
}
